tracetool/Modules/libva.py

```diff
@@ -0,0 +1,596 @@
+        data['Buffers'] = info
+
+    def parseBuffer(self, data):
+        raw = bytes.fromhex(data['BufData'])
+        profile = GetEnumName(data['Profile'])
+        type = GetEnumName(data['BufferType'])
+        cnt = data['MemSize']//data['Size']
+        ret = {'Profile':profile, 'BufferType':type, 'Count':cnt, 'Size':data['Size'], 'Details':[]}
+        if profile not in self.profileMap:
+            print('Warning! non supported profile {s}'.format(profile))
+            return None
+        if profile not in self.dispatch:
+            self.dispatch[profile] = self.profileMap[profile]()
+        pos = 0
+        for i in range(cnt):
+            out = self.dispatch[profile].parse(type, raw[pos:(i+1)*data['Size']])
+            if out == None:
+                print('Warning! non supported buffer type {s}'.format(type))
+                return None
+            ret['Details'].append(out)
+            pos += data['Size']
+        return ret
+
+class buffer(Structure):
+    def formatArray(self, val):
+        ret = ''
+        if val._type_ is c_ubyte:
+            ptr = cast(val, POINTER(c_ubyte))
+        elif val._type_ is c_ushort:
+            ptr = cast(val, POINTER(c_ushort))
+        elif val._type_ is c_uint:
+            ptr = cast(val, POINTER(c_uint))
+        else:
+            print('Warning! non supported array type {}'.format(val._type_))
+            return ret
+
+        for i in range(len(val)):
+            ret += '{:02x} '.format(ptr[i])
+            if i & 15 == 15:
+                ret += '\n'
+        return ret
+
+    def getDict(self):
+        ret = {}
+        for field in self._fields_:
+            val = getattr(self, field[0])
+            if isinstance(val, buffer):
+                ret[field[0]] = val.getDict()
+                continue
+            if isinstance(val, Array):
+                # buffer array
+                if isinstance(val[0], buffer):
+                    out = []
+                    for i in val:
+                        out.append(i.getDict())
+                    ret[field[0]] = out
+                    continue
+                # 2d array
+                if isinstance(val[0], Array):
+                    out = []
+                    for i in val:
+                        out.append(self.formatArray(i))
+                    ret[field[0]] = out
+                    continue
+                ret[field[0]] = self.formatArray(val)
+                continue
+            ret[field[0]] = val
+        return ret
+
+class base:
+    def parse(self, type, raw):
+        if type in self.dispatch:
+            return self.dispatch[type](raw)
+        return self.formatData(raw)
+
+    def formatData(self, raw):
+        ret = ''
+        i = 0
+        while i <len(raw):
+            ret += '{:02x} '.format(raw[i])
+            i += 1
+            if i & 15 == 15:
+                ret += '\n'
+        return ret
+
+class mpeg2(base):
+    def __init__(self):
+        self.dispatch = {
```

Comment:
only add mpeg2 decode and h264 encode buffer data support at first version.
try run and capture trace with libva-util mpeg2 decode and h264 encode sample app.